homeassistant/components/progettihwsw/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for ProgettiHWSW Automation integration."""
+
+from ProgettiHWSW.ProgettiHWSWAPI import ProgettiHWSWAPI
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = vol.Schema({vol.Required("host"): str})
+
+
+class EntryValidator:
+    """Class to validate data given by user."""
+
+    def __init__(self, host):
+        """Initialize the data."""
+        self._host = host
+        self._phwsw_instance = ProgettiHWSWAPI(host)
+
+    async def check_board_validity(self, hass):
```

Comment:
can be sync per below comment

Reply:
I made it sync and used executor to call it from async validate_input function.